x/stakeibc/keeper/reward_converter.go

```diff
@@ -53,6 +59,124 @@ type ForwardMetadata struct {
+// For instance, with dYdX, this is applied to the DYDX tokens that were converted from USDC
+//
+// If the chain doesn't have a rebate in place, the split is decided entirely from the stride commission percent
+// However, if the chain does have a rebate, we need to factor that into the calculation, by scaling
+// up the rewards to find the amount before the rebate
+//
+// For instance, if 1000 rewards were collected and 2 were sent as a rebate, then the stride fee should be based
+// on the original 1000 rewards instead of the remaining 998 in the query response:
+//
+//	Community pool liquid staked 1M, TVL is 10M, rebate is 20%, stride fee is 10%
+//	If 998 native tokens were queried, we have to scale that up to 1000 original reward tokens
+//
+//	Effective Rebate Pct = 10% fees * (1M LS / 10M TVL) * 20% rebate = 0.20% (aka 0.002)
+//	Effective Stride Fee Pct = 10% fees - 0.20% effective rebate = 9.8%
+//	Original Reward Amount = 998 Queried Rewards / (1 - 0.002 effective rebate rate) = 1000 original rewards
+//	Then stride fees are 9.8% of that 1000 original rewards = 98
+func (k Keeper) CalculateRewardsSplitAfterRebate(
+	ctx sdk.Context,
+	hostZone types.HostZone,
+	rewardsAmount sdkmath.Int,
+) (strideFeeAmount sdkmath.Int, reinvestAmount sdkmath.Int, err error) {
+	// Get the fee rate and total fees from params (e.g. 0.1 for 10% fee)
+	strideFee := sdk.NewIntFromUint64(k.GetParams(ctx).StrideCommission)
+	totalFeeRate := sdk.NewDecFromInt(strideFee).Quo(sdk.NewDec(100))
+
+	// Check if the chain has a rebate
+	rebateInfo, chainHasRebate := hostZone.SafelyGetCommunityPoolRebate()
+
+	// If there's no rebate, the fee split just uses the commission
+	if !chainHasRebate {
+		strideFeeAmount = sdk.NewDecFromInt(rewardsAmount).Mul(totalFeeRate).TruncateInt()
+	} else {
+		// It shouldn't be possible to have 0 delegations (since there are rewards and there was a community pool stake)
+		// This will also prevent a division by 0 error
+		if hostZone.TotalDelegations.IsZero() {
+			return sdkmath.ZeroInt(), sdkmath.ZeroInt(), errorsmod.Wrapf(types.ErrDivisionByZero,
+				"unable to calculate rebate amount for %s since total delegations are 0", hostZone.ChainId)
+		}
+
+		// It also shouldn't be possible for the liquid stake amount to be greater than the full TVL
+		if rebateInfo.LiquidStakeAmount.GT(hostZone.TotalDelegations) {
+			return sdkmath.ZeroInt(), sdkmath.ZeroInt(), errorsmod.Wrapf(types.ErrFeeSplitInvariantFailed,
+				"community pool liquid staked amount greater than total delegations")
+		}
+
+		// Otherwise, the rebate must be considered in the fee split
+		// The rebate portion is the portion of TVL contributed to by the liquid stake * the rebate percentage
+		// The stride fee poriton is the remaining percentage
+		contributionRate := sdk.NewDecFromInt(rebateInfo.LiquidStakeAmount).Quo(sdk.NewDecFromInt(hostZone.TotalDelegations))
```

Comment:
Small thing to note: the `TotalDelegations` here is being fetched in a different epoch than when `CalculateRewardsSplitBeforeRebate` is called. Generally, the two values should be very close together, and I don't think there are any bad edge cases, but just verifying that lines up with your intuition too

Reply:
yeah my thinking was that it's not perfect, but the values are close enough together and it prevents us from having to make this stateful. 

Also, fwiw, this was changed in the other PR to be the stTokens and total supply (similar story there though)